compiler warnings with LibreSSL 3.7.3

First warning is because `SSL_CIPHER_get_id` returns an `unsigned long` with LibreSSL (also in latest 3.8.1) and `uint32_t` with OpenSSL/quictls (latest 3.1.2) and BoringSSL.

The second one is due to an extra function pointer `set_read_secret`. This is missing from OpenSSL/quictls (latest 3.1.2), but present in LibreSSL (also in latest 3.8.1) and BoringSSL.

```
Linux llvm 16 and Apple clang 14:
ngtcp2/crypto/quictls/quictls.c:195:15: warning: implicit conversion loses integer precision: 'unsigned long' to 'uint32_t' (aka 'unsigned int') [-Wshorten-64-to-32]
ngtcp2/crypto/quictls/quictls.c:824:1: warning: missing field 'set_read_secret' initializer [-Wmissing-field-initializers]

mingw-64 llvm 16:
ngtcp2/crypto/quictls/quictls.c:824:1: warning: missing field 'set_read_secret' initializer [-Wmissing-field-initializers]

gcc 9:
ngtcp2/crypto/quictls/quictls.c:195:15: warning: conversion from 'long unsigned int' to 'uint32_t' {aka 'unsigned int'} may change value [-Wconversion]
ngtcp2/crypto/quictls/quictls.c:824:1: warning: missing initializer for field 'set_read_secret' of 'SSL_QUIC_METHOD' {aka 'struct ssl_quic_method_st'} [-Wmissing-field-initializers]
```

--- a/crypto/quictls/quictls.c
+++ b/crypto/quictls/quictls.c
@@ -192,7 +192,7 @@ ngtcp2_crypto_ctx *ngtcp2_crypto_ctx_tls(ngtcp2_crypto_ctx *ctx,
     return NULL;
   }
 
-  cipher_id = SSL_CIPHER_get_id(cipher);
+  cipher_id = (uint32_t)SSL_CIPHER_get_id(cipher);
 
   if (!supported_cipher_id(cipher_id)) {
     return NULL;
@@ -821,6 +821,10 @@ static SSL_QUIC_METHOD quic_method = {
     add_handshake_data,
     flush_flight,
     send_alert,
+#ifdef LIBRESSL_VERSION_NUMBER
+    NULL,
+    NULL,
+#endif /* LIBRESSL_VERSION_NUMBER */
 };
 
 static void crypto_quictls_configure_context(SSL_CTX *ssl_ctx) {

